jujugui/static/gui/src/app/components/post-deployment/post-deployment.js

```diff
@@ -0,0 +1,204 @@
+/* Copyright (C) 2015 Canonical Ltd. */
+
+'use strict';
+
+/**
+  Displays post-deployment information for usage and quickstart.
+  Shown when a charm or bundle is added to the canvas, and on deploy.
+*/
+class PostDeployment extends React.Component {
+  constructor(props) {
+    super(props);
+    this.variables = {
+      details_link: {
+        onClick: this._handleViewDetails.bind(this),
+        content: '<span role="button" ' +
+          'class="link" ' +
+          'data-variable="details_link">View details</span>'
+      },
+      requires_cli_link: {
+        content: '<a ' +
+          'href="https://jujucharms.com/docs/stable/reference-install" ' +
+          'target="_blank">Juju CLI client</a>'
+      }
+    };
+    this.state= {
+      content: null,
+      metadata: {}
+    };
+  }
+
+  componentDidMount() {
```

Comment:
If we want to reduce requests to the archive endpoint another solution may be to make a request to `getEntity` it will include with it the `manifest` which is the list of files. In there you'll be able to check for the `usage.md` and then request it as you have done here.

https://github.com/juju/juju-gui/blob/develop/jujugui/static/gui/src/app/jujulib/charmstore.js#L339

This will of course require two requests so it really depends how demanding this is going to be...